### PR TITLE
Reduce daylighting allocations

### PR DESCRIPTION
**Pull request by mjwitte**
Pull request overview
---------------------
While the daylighting code is fresh in mind, an attempt at some small performance improvement by avoiding local allocates and reducing the size of arrays when there are no blinds in the model.

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [ ] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [ ] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [ ] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [ ] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [ ] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [ ] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

**Comment by mjwitte** [comment]
This includes the changes from #9002 so it looks like more changes than it really is.

File: src/EnergyPlus/DataDaylighting.hh [hunk 1]
```diff
@@ -95,88 +96,89 @@ namespace DataDaylighting {
...
-    // Parameters for "Lighting Control Type"
-    enum class iLtgCtrlType
+    // Parameters for "Lighting Control Type" - these are the values expected by DElight
+    enum class LtgCtrlType
     {
+        Invalid = 0,
         Continuous = 1,
         Stepped = 2,
-        ContinuousOff = 3
+        ContinuousOff = 3,
```
Comment (rraustad) [inline src/EnergyPlus/DataDaylighting.hh]: Why was a comma added here? Does unassigned = -1 apply here with num at the end? I see a lot of comments on that, and yeah it's a good practice, but is there any overhead for those enums where num is never used?

Comment (amirroth) [reply src/EnergyPlus/DataDaylighting.hh]: Enumerations are compile-time constants, not runtime variables. There is no storage or other runtime overhead associated with them. [Here](https://github.com/NREL/EnergyPlus/wiki/enum-or-enum-class:-Either-Way,-Use-Them) is a more detailed explanation of the patterns we are trying to establish.

**Comment by mjwitte** [comment]
@Myoldmopar After #9002 is merged, I'll pull this branch up and undraft it for possible inclusion in 9.6, but no big deal if it misses the boat.

**Comment by mjwitte** [comment]
@Myoldmopar This is ready for review. No big deal if it waits until post-9.6

**Comment by Myoldmopar** [comment]
This one looks good, no need to wait.  I'm currently building and testing with both this and #9065 merged in and if all is good they will both drop.

**Comment by Myoldmopar** [comment]
Everything passes locally with both this and the other PR in.  Merging now.

**Pull request merged**